File: CS/Reporting_how-to-programmatically-create-a-table-and-manually-add-rows-and-cells-to-it-e1333/XtraReport1.cs
using System.Drawing;
using System.Drawing.Printing;
using DevExpress.XtraReports.UI;
// ...

namespace Controls_XRTable {
    public partial class XtraReport1 : XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void XtraReport1_BeforePrint(object sender, PrintEventArgs e) {
            this.Detail.Controls.Add(CreateXRTable());
        }

        public XRTable CreateXRTable() {
            // Create an empty table and set its size.
            XRTable table = new XRTable();
            table.Size = new Size(300, 100);

            // Start table initialization.
            table.BeginInit();

            // Enable table borders to see its boundaries.
            table.BorderWidth = 2;
            table.Borders = DevExpress.XtraPrinting.BorderSide.All;

            // Create a table row.
            XRTableRow row = new XRTableRow();

            // Create two table cells.
            XRTableCell cell0 = new XRTableCell();
            XRTableCell cell1 = new XRTableCell();

            // Construct the table.
            row.Cells.Add(cell0);
            row.Cells.Add(cell1);
            table.Rows.Add(row);

            // Finish table initialization.
            table.EndInit();

            return table;
        }


    }
}

File: VB/Reporting_how-to-programmatically-create-a-table-and-manually-add-rows-and-cells-to-it-e1333/XtraReport1.vb
Imports System.Drawing
Imports System.Drawing.Printing
Imports DevExpress.XtraReports.UI
' ...

Namespace Controls_XRTable
	Partial Public Class XtraReport1
		Inherits XtraReport

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub XtraReport1_BeforePrint(ByVal sender As Object, ByVal e As PrintEventArgs) Handles Me.BeforePrint
			Me.Detail.Controls.Add(CreateXRTable())
		End Sub

		Public Function CreateXRTable() As XRTable
			' Create an empty table and set its size.
			Dim table As New XRTable()
			table.Size = New Size(300, 100)

			' Start table initialization.
			table.BeginInit()

			' Enable table borders to see its boundaries.
			table.BorderWidth = 2
			table.Borders = DevExpress.XtraPrinting.BorderSide.All

			' Create a table row.
			Dim row As New XRTableRow()

			' Create two table cells.
			Dim cell0 As New XRTableCell()
			Dim cell1 As New XRTableCell()

			' Construct the table.
			row.Cells.Add(cell0)
			row.Cells.Add(cell1)
			table.Rows.Add(row)

			' Finish table initialization.
			table.EndInit()

			Return table
		End Function


	End Class
End Namespace
